io/zenoh-transport/src/multicast/establishment.rs

```diff
@@ -11,75 +11,81 @@
 // Contributors:
 //   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
 //
-use crate::multicast::TransportMulticast;
-use crate::TransportManager;
-// use rand::Rng;
-// use std::sync::Arc;
-// use zenoh_core::{zasynclock, zlock};
+use crate::{
+    common::seq_num,
+    multicast::{transport::TransportMulticastInner, TransportMulticast},
+    TransportConfigMulticast, TransportManager,
+};
+use rand::Rng;
+use std::sync::Arc;
+use zenoh_core::zasynclock;
 use zenoh_link::LinkMulticast;
-// use zenoh_protocol::core::{ConduitSn, ConduitSnList, Priority};
+use zenoh_protocol::{
+    core::{Field, Priority},
+    transport::PrioritySn,
+};
 use zenoh_result::ZResult;
 
 pub(crate) async fn open_link(
-    _manager: &TransportManager,
-    _link: LinkMulticast,
+    manager: &TransportManager,
+    link: LinkMulticast,
 ) -> ZResult<TransportMulticast> {
-    todo!();
-    // // Create and configure the multicast transport
-    // let mut prng = zasynclock!(manager.prng);
+    // Create and configure the multicast transport
+    let mut prng = zasynclock!(manager.prng);
```

Comment:
This lock is held until end of function, shouldn't it be released at line 54?

Reply:
Good catch.